lib/inbox.rb

```diff
@@ -114,11 +114,11 @@ def url_for_path(path)
     end
 
     def url_for_authentication(redirect_uri, login_hint = '', options = {})
-      trialString = 'false'
-      if options[:trial] == true
-        trialString = 'true'
-      end
-      "https://#{@service_domain}/oauth/authorize?client_id=#{@app_id}&trial=#{trialString}&response_type=code&scope=email&login_hint=#{login_hint}&redirect_uri=#{redirect_uri}"
+      arguments = {
+        trial: false,
+        login_hint: login_hint
+      }.merge(options)
+      URI::HTTPS.build(host: @service_domain, query: arguments.to_query).to_s
```

Comment:
As it is, this won't work as it doesn't include any of the `path` portion, and several of the required `query` arguments are also gone.

That said, I totally agree that [using `URI` or `Addressable` would be much preferred](https://github.com/nylas/nylas-ruby/pull/38#discussion_r39870243) to hand-rolling.